File: 백준/Silver/11501. 주식/주식.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

let idx = 0;
const T = parseInt(input[idx++]);

for (let t = 0; t < T; t++) {
    const N = parseInt(input[idx++]);
    const ls = input[idx++].split(" ").map(Number);

    let num = 0;
    let ans = 0;

    for (let i = N - 1; i >= 0; i--) {
        if (ls[i] > num) {
            num = ls[i];
        } else {
            ans += num - ls[i];
        }
    }

    console.log(ans);
}

File: 백준/Silver/11501. 주식/주식.py
T = int(input())
for _ in range(T):
    N = int(input())
    ls = list(map(int, input().split()))
    num = 0
    ans = 0

    for i in range(N-1,-1,-1):
        if ls[i] > num:
            num = ls[i]
        else:
            ans += (num-ls[i])
    print(ans)